### PR TITLE
Error Prone 2.21.0, NullableOptional check

### PR DESCRIPTION
**Pull request by ksobolew**
## Description

* Bump Error Prone to 2.21.0
* Enable check: `NullableOptional`

## Additional context and related issues

The check enabled is a new one in that release, and it looks like it complements `NullOptional` pretty nicely.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2474,6 +2474,7 @@
                                     -Xep:MissingSummary:OFF \
                                     -Xep:MutablePublicArray:ERROR \
                                     -Xep:NarrowingCompoundAssignment:ERROR \
+                                    -Xep:NullableOptional:ERROR \
                                     -Xep:NullOptional:ERROR \
```
Comment (findepi) [inline pom.xml:2478]: is the sorting case-sensitive or case-insensitive?
for example, both vim and Intellij's "Sort lines" sort these lines differently

Comment (ksobolew) [reply pom.xml:2478]: I was wondering about it too, but I always find it confusing, so I usually don't bother. Reordering is not a problem. We can definitely decide that "whatever the tools are doing" is the policy here :)

Comment (ksobolew) [reply pom.xml:2477]: Reordered

**Review by wendigo — APPROVED** [review]

**Comment by wendigo** [comment]
Thx @ksobolew

**Pull request merged**